### ISSUE TITLE
SIM401 should catch ternary operations

### ISSUE THREAD

**Issue by jerr0328** [astral-sh/ruff#7288]
In #4932, there was a bug because if-else blocks were rewritten as ternary operations which SIM401 wouldn't be able to fix. The bug was fixed but there's still the fact that you can have code like

```python
obj: dict = ...
key = ...

value = obj[key] if key in obj else "Not found"
```

which will not be corrected with `ruff check --fix --select SIM .`

Expected behavior:
```python
obj: dict = ...
key = ...

value = obj.get(key, "Not found")
``` 

Checked with ruff `v0.0.287`

**Comment by charliermarsh** [comment]
Makes sense to me.

**Comment by Flowrey** [comment]
Hey, I would like to give it a try can I ?

**Comment by zanieb** [comment]
@Flowrey feel free!

**Comment by Flowrey** [comment]
I'm a bit puzzled, ternary operations are expressions, `Expr::IfExp`, but `if-else-block-instead-of-dict-get` rule is defined for statement and expect an `StmtIf` as argument, so should i define a new rule in this case like `if-expr-instead-of-dict-get` ?

**Comment by charliermarsh** [comment]
I think it can be the same _rule_, but it will need to be a new function that takes an `Expr::IfExp`, and is called from `crates/ruff/src/checkers/ast/analyze/expression.rs`. We may also need to rename the rule to reflect the two cases.
